### PR TITLE
make core.config self-contained

### PR DESCRIPTION
**Pull request by jbrockmendel**
Broken off from #25176

### CODE REVIEW

File: pandas/core/config.py
Comment (WillAyd) [inline pandas/core/config.py]: Assuming this is targeted for 0.25 do we even need to bother with this?

Comment (jbrockmendel) [reply pandas/core/config.py]: I figure until we actually start ripping out compat code (any day now I hope) its easier to leave this in and keep the ball rolling on these

**Comment by jreback** [comment]
prob simpler to wait till we start ripping PY2 out, soon, just as release 0.24.2

**Comment by jbrockmendel** [comment]
Another option: `unicode` is only used to define `is_unicode`, which is never used anywhere.  Could just remove both of em.

File: pandas/core/config.py
Comment (gfyoung) [reply pandas/core/config.py]: We could just wait until `0.24.2` comes out before merging this.

**Comment by jbrockmendel** [comment]
Just rebased.  Removed the compat code, since it was only necessary for `is_unicode`, which is never used anywhere.  Should be good to go.

File: pandas/core/config.py
Comment (WillAyd) [inline pandas/core/config.py]: Can get rid of the u prefix now, no?

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/core/config.py
Comment (jbrockmendel) [reply pandas/core/config.py]: Are we "full speed ahead" on ripping out py2?  If not, let's hold off.  I'd like to get this and #25757 merged since there are nice follow-ups.

Comment (jreback) [reply pandas/core/config.py]: yes, you can remove PY2 things

Comment (jbrockmendel) [reply pandas/core/config.py]: OK, just pushed.

**Comment by jreback** [comment]
lgtm. merge master ping on green.

**Comment by jbrockmendel** [comment]
ping

**Comment by jreback** [comment]
thanks!

**Pull request merged**
